NPE when disabling OpenTelemetry

### Describe the bug

Something observed on the nightly Camel Quarkus builds with Quarkus 999-SNAPSHOT. We have a test where otel is disabled with `quarkus.otel.enabled = false`. The application fails to start due to:

```
java.lang.RuntimeException: Failed to start quarkus
    at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
    at io.quarkus.runtime.Application.start(Application.java:101)
    at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
    at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
    at io.quarkus.runner.GeneratedMain.main(Unknown Source)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.base/java.lang.reflect.Method.invoke(Method.java:568)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
    at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: java.lang.NullPointerException: Cannot invoke "io.smallrye.config.ConfigValue.getValue()" because "value" is null
    at io.quarkus.opentelemetry.runtime.config.OTelFallbackConfigSourceInterceptor.getValue(OTelFallbackConfigSourceInterceptor.java:46)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
    at io.smallrye.config.SecretKeysHandlerConfigSourceInterceptor.getValue(SecretKeysHandlerConfigSourceInterceptor.java:26)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
    at io.smallrye.config.FallbackConfigSourceInterceptor.getValue(FallbackConfigSourceInterceptor.java:24)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
    at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:382)
    at io.quarkus.runtime.configuration.ConfigRecorder.handleConfigChange(ConfigRecorder.java:43)
    at io.quarkus.deployment.steps.ConfigGenerationBuildStep$checkForBuildTimeConfigChange1532146938.deploy_4(Unknown Source)
    at io.quarkus.deployment.steps.ConfigGenerationBuildStep$checkForBuildTimeConfigChange1532146938.deploy(Unknown Source)
    ... 13 more
```

I see the same thing if I generate a basic app (E.g just with quarkus-otel & no Camel involved) from [code.quarkus.io](http://code.quarkus.io/) and tweak the Quarkus version to 999-SNAPSHOT.


### Expected behavior

OpenTelemetry is disabled as expected without any issues.

### Actual behavior

An NPE is thrown (see above stack trace).

### How to Reproduce?

1. Create an empty application
2. Set the Quarkus version to `999-SNAPSHOT`
3. Add a dependency for `quarkus-opentelemetry`
4. Disable otel in àpplication.properties` with `quarkus.otel.enabled = false`
5. Package the application and execute the runnable JAR

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/OTelFallbackConfigSourceInterceptor.java
+++ b/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/OTelFallbackConfigSourceInterceptor.java
@@ -42,7 +42,7 @@ public class OTelFallbackConfigSourceInterceptor extends FallbackConfigSourceInt
     @Override
     public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
         ConfigValue value = super.getValue(context, name);
-        if (name.equals("quarkus.otel.traces.sampler")) {
+        if (value != null && name.equals("quarkus.otel.traces.sampler")) {
             return value.withValue(LEGACY_SAMPLER_NAME_CONVERTER.convert(value.getValue()));
         }
         return value;

--- a/integration-tests/opentelemetry-quickstart/pom.xml
+++ b/integration-tests/opentelemetry-quickstart/pom.xml
@@ -1,0 +1,188 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-integration-tests-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>quarkus-integration-test-opentelemetry-quickstart</artifactId>
+    <name>Quarkus - Integration Tests - OpenTelemetry quickstart</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-opentelemetry</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive</artifactId>
+        </dependency>
+
+
+        <!-- Needed for InMemorySpanExporter to verify captured traces -->
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-sdk-testing</artifactId>
+        </dependency>
+
+        <!-- Test Dependencies -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <!-- Minimal test dependencies to *-deployment artifacts for consistent build order -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive-jackson-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-opentelemetry-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables>
+                        <vertx.disableWebsockets>false</vertx.disableWebsockets>
+                    </systemPropertyVariables>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <id>native-image</id>
+            <activation>
+                <property>
+                    <name>native</name>
+                </property>
+            </activation>
+
+            <properties>
+                <quarkus.package.type>native</quarkus.package.type>
+            </properties>
+
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skipTests>${native.surefire.skip}</skipTests>
+                            <systemPropertyVariables>
+                                <vertx.disableWebsockets>false</vertx.disableWebsockets>
+                            </systemPropertyVariables>
+                        </configuration>
+                    </plugin>
+
+                    <plugin>
+                        <artifactId>maven-failsafe-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>integration-test</goal>
+                                    <goal>verify</goal>
+                                </goals>
+                                <configuration>
+                                    <systemPropertyVariables>
+                                        <vertx.disableWebsockets>false</vertx.disableWebsockets>
+                                        <native.image.path>
+                                            ${project.build.directory}/${project.build.finalName}-runner
+                                        </native.image.path>
+                                    </systemPropertyVariables>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+</project>

--- a/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/ExporterResource.java
+++ b/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/ExporterResource.java
@@ -1,0 +1,46 @@
+package io.quarkus.it.opentelemetry;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Response;
+
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter;
+import io.opentelemetry.sdk.trace.data.SpanData;
+
+@Path("")
+public class ExporterResource {
+    @Inject
+    InMemorySpanExporter inMemorySpanExporter;
+
+    @GET
+    @Path("/reset")
+    public Response reset() {
+        inMemorySpanExporter.reset();
+        return Response.ok().build();
+    }
+
+    @GET
+    @Path("/export")
+    public List<SpanData> export() {
+        return inMemorySpanExporter.getFinishedSpanItems()
+                .stream()
+                .filter(sd -> !sd.getName().contains("export") && !sd.getName().contains("reset"))
+                .collect(Collectors.toList());
+    }
+
+    @ApplicationScoped
+    static class InMemorySpanExporterProducer {
+        @Produces
+        @Singleton
+        InMemorySpanExporter inMemorySpanExporter() {
+            return InMemorySpanExporter.create();
+        }
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/GreetingResource.java
+++ b/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/GreetingResource.java
@@ -1,0 +1,16 @@
+package io.quarkus.it.opentelemetry;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "Hello from RESTEasy Reactive";
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/output/SpanDataModuleSerializer.java
+++ b/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/output/SpanDataModuleSerializer.java
@@ -1,0 +1,19 @@
+package io.quarkus.it.opentelemetry.output;
+
+import jakarta.inject.Singleton;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.quarkus.jackson.ObjectMapperCustomizer;
+
+@Singleton
+public class SpanDataModuleSerializer implements ObjectMapperCustomizer {
+    @Override
+    public void customize(ObjectMapper objectMapper) {
+        SimpleModule simpleModule = new SimpleModule();
+        simpleModule.addSerializer(SpanData.class, new SpanDataSerializer());
+        objectMapper.registerModule(simpleModule);
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/output/SpanDataSerializer.java
+++ b/integration-tests/opentelemetry-quickstart/src/main/java/io/quarkus/it/opentelemetry/output/SpanDataSerializer.java
@@ -1,0 +1,55 @@
+package io.quarkus.it.opentelemetry.output;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import io.opentelemetry.sdk.trace.data.SpanData;
+
+public class SpanDataSerializer extends StdSerializer<SpanData> {
+    public SpanDataSerializer() {
+        this(null);
+    }
+
+    public SpanDataSerializer(Class<SpanData> type) {
+        super(type);
+    }
+
+    @Override
+    public void serialize(SpanData spanData, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+            throws IOException {
+        jsonGenerator.writeStartObject();
+
+        jsonGenerator.writeStringField("spanId", spanData.getSpanId());
+        jsonGenerator.writeStringField("traceId", spanData.getTraceId());
+        jsonGenerator.writeStringField("name", spanData.getName());
+        jsonGenerator.writeStringField("kind", spanData.getKind().name());
+        jsonGenerator.writeBooleanField("ended", spanData.hasEnded());
+
+        jsonGenerator.writeStringField("parentSpanId", spanData.getParentSpanContext().getSpanId());
+        jsonGenerator.writeStringField("parent_spanId", spanData.getParentSpanContext().getSpanId());
+        jsonGenerator.writeStringField("parent_traceId", spanData.getParentSpanContext().getTraceId());
+        jsonGenerator.writeBooleanField("parent_remote", spanData.getParentSpanContext().isRemote());
+        jsonGenerator.writeBooleanField("parent_valid", spanData.getParentSpanContext().isValid());
+
+        spanData.getAttributes().forEach((k, v) -> {
+            try {
+                jsonGenerator.writeStringField("attr_" + k.getKey(), v.toString());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+
+        spanData.getResource().getAttributes().forEach((k, v) -> {
+            try {
+                jsonGenerator.writeStringField("resource_" + k.getKey(), v.toString());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+
+        jsonGenerator.writeEndObject();
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/main/resources/META-INF/resources/test.html
+++ b/integration-tests/opentelemetry-quickstart/src/main/resources/META-INF/resources/test.html
@@ -1,0 +1,1 @@
+<html><body>Test</body></html>

--- a/integration-tests/opentelemetry-quickstart/src/main/resources/application.properties
+++ b/integration-tests/opentelemetry-quickstart/src/main/resources/application.properties
@@ -1,0 +1,3 @@
+# speed up build
+quarkus.otel.bsp.schedule.delay=0
+quarkus.otel.bsp.export.timeout=5s

--- a/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/BaseTest.java
+++ b/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/BaseTest.java
@@ -1,0 +1,20 @@
+package io.quarkus.it.opentelemetry;
+
+import static io.restassured.RestAssured.get;
+
+import java.util.List;
+import java.util.Map;
+
+import io.restassured.common.mapper.TypeRef;
+
+public class BaseTest {
+
+    protected List<Map<String, Object>> getSpans() {
+        return get("/export").body().as(new TypeRef<>() {
+        });
+    }
+
+    protected void buildGlobalTelemetryInstance() {
+        // Do nothing in JVM mode
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryDisabledIT.java
+++ b/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryDisabledIT.java
@@ -1,0 +1,19 @@
+package io.quarkus.it.opentelemetry;
+
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.OpenTelemetrySdkBuilder;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+
+@QuarkusIntegrationTest
+public class OpenTelemetryDisabledIT extends OpenTelemetryDisabledTest {
+    @Override
+    protected void buildGlobalTelemetryInstance() {
+        // When running native tests the test class is outside the Quarkus application,
+        // so we need to set the propagator on the GlobalOpenTelemetry instance
+        OpenTelemetrySdkBuilder builder = OpenTelemetrySdk.builder();
+        builder.setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()));
+        builder.buildAndRegisterGlobal();
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryDisabledTest.java
+++ b/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryDisabledTest.java
@@ -1,0 +1,39 @@
+package io.quarkus.it.opentelemetry;
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.is;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+
+@QuarkusTest
+@TestProfile(OpenTelemetryDisabledTest.MyProfile.class)
+public class OpenTelemetryDisabledTest extends BaseTest {
+
+    @Test
+    void buildTimeDisabled() {
+        given()
+                .when().get("/hello")
+                .then()
+                .statusCode(200)
+                .body(is("Hello from RESTEasy Reactive"));
+        // Service will start nevertheless.
+        await().atMost(200, MILLISECONDS).until(() -> getSpans().size() == 0);
+    }
+
+    public static class MyProfile implements QuarkusTestProfile {
+
+        @Override
+        public Map<String, String> getConfigOverrides() {
+            return Map.of("quarkus.otel.enabled", "false");
+        }
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryIT.java
+++ b/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryIT.java
@@ -1,0 +1,19 @@
+package io.quarkus.it.opentelemetry;
+
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.OpenTelemetrySdkBuilder;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+
+@QuarkusIntegrationTest
+public class OpenTelemetryIT extends OpenTelemetryTest {
+    @Override
+    protected void buildGlobalTelemetryInstance() {
+        // When running native tests the test class is outside the Quarkus application,
+        // so we need to set the propagator on the GlobalOpenTelemetry instance
+        OpenTelemetrySdkBuilder builder = OpenTelemetrySdk.builder();
+        builder.setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()));
+        builder.buildAndRegisterGlobal();
+    }
+}

--- a/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTest.java
+++ b/integration-tests/opentelemetry-quickstart/src/test/java/io/quarkus/it/opentelemetry/OpenTelemetryTest.java
@@ -1,0 +1,24 @@
+package io.quarkus.it.opentelemetry;
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.is;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class OpenTelemetryTest extends BaseTest {
+    @Test
+    void buildTimeEnabled() {
+        given()
+                .when().get("/hello")
+                .then()
+                .statusCode(200)
+                .body(is("Hello from RESTEasy Reactive"));
+        await().atMost(5, SECONDS).until(() -> getSpans().size() == 1);
+    }
+}

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -373,6 +373,7 @@
                 <module>micrometer-mp-metrics</module>
                 <module>micrometer-prometheus</module>
                 <module>opentelemetry</module>
+                <module>opentelemetry-quickstart</module>
                 <module>opentelemetry-spi</module>
                 <module>opentelemetry-jdbc-instrumentation</module>
                 <module>opentelemetry-quartz</module>

